xrmocap/core/estimation/mview_sperson_smpl_estimator.py

```diff
@@ -401,10 +421,32 @@ def estimate_smpl(self, keypoints3d: Keypoints) -> SMPLData:
                 keypoints3d_conf=kps3d_conf,
                 keypoints3d_convention=working_convention,
                 handler_key='keypoints3d_limb_len'))
+
         registrant_output = self.smplify(
-            input_list=[kp3d_mse_input, kp3d_llen_input])
-        smpl_data = SMPLData()
+            input_list=[kp3d_mse_input, kp3d_llen_input],
+            init_param_dict=init_smpl_dict,
+            return_joints=return_joints,
+            return_verts=return_verts)
+
+        if self.smpl_data_type == 'smplx':
+            smpl_data = SMPLXData()
+        elif self.smpl_data_type == 'smpl':
+            smpl_data = SMPLData()
+        else:
+            self.logger.warning('smpl data type not defined, '
+                                'set to smpl by default')
+            smpl_data = SMPLData()
         smpl_data.from_param_dict(registrant_output)
+
+        if return_joints or return_verts:
+            adhoc_data = {}
+            if return_joints:
+                adhoc_data['joints'] = registrant_output['joints']
+            if return_verts:
+                adhoc_data['vertices'] = registrant_output['vertices']
+
+            return smpl_data, adhoc_data
+
         return smpl_data
```

Comment:
We've found that it is dangerous if sometime we return one value, other time we do not.
```suggestion
        return smpl_data, None
```

Reply:
Change to return 2 values by default may affect some existing usages when they update, shall we consider this as well?

Reply:
The other solution is that:
```python
smpl_data['vertices'] = registrant_output['vertices']
```
And take vertices out of smpl_data as soon as we call estimate smpl. SMPLData is capable of containing the `adhoc_data` as its super class `dict`, but it cannot maintain it forever.

Reply:
That's a better solution and I think we need some reminders here if we do so. If users do `smpl_data.dump()` or `np.savez(smpl_data)`, without popping adhoc_data,  it will be saved and even loaded from files later, which is not desired to have `joints` and `vertices` in `SMPLData` class. It won't return through `to_param_dict()`, but I don't think `adhoc_data` will be auto-removed from `smpl_data` once we set them in the `dict`.

Reply:
Fixed in [21a9b15](https://github.com/openxrlab/xrmocap/pull/52/commits/21a9b1598c19c0618ff463203501f7afeea2e43d).